src/scenarios/shared/mauisharedpython.py

```diff
@@ -1,24 +1,98 @@
-import subprocess
+import json
 import os
-import requests
+import xml.etree.ElementTree as ET
+import re
+import urllib.request
+from logging import getLogger
+from performance.common import get_repo_root_path
 from shared.precommands import PreCommands
 
 # Remove the aab files as we don't need them, this saves space in the correlation payload
 def remove_aab_files(output_dir="."):
     file_list = os.listdir(output_dir)
     for file in file_list:
         if file.endswith(".aab"):
-            os.remove(os.path.join(output_dir, file))
+            os.remove(os.path.join(output_dir, file))  
+
+def generate_maui_rollback_dict():
+    # Generate and use rollback based on Version.Details.xml
+    # Generate the list of versions starts to get and the names to save them as in the rollback.
+    # These mapping values were taken from the previously generated rollback files for the maui workload. There should be at least one entry for each
+    # of the Maui Workload dependencies in the /eng/Version.Details.xml file, aside from VS.Tools.Net.Core.SDK.Resolver.
+    # If there are errors in the future, reach out to the maui team.
+    rollback_name_to_xml_name_mappings: dict[str, str] = {
+        "microsoft.net.sdk.android" : "Microsoft.Android.Sdk",
+        "microsoft.net.sdk.ios" : "Microsoft.iOS.Sdk",
+        "microsoft.net.sdk.maccatalyst" : "Microsoft.MacCatalyst.Sdk",
+        "microsoft.net.sdk.macos" : "Microsoft.macOS.Sdk",
+        "microsoft.net.sdk.maui" : "Microsoft.Maui.Controls",
+        "microsoft.net.sdk.tvos" : "Microsoft.tvOS.Sdk",
+        "microsoft.net.sdk.mono.toolchain.current" : "Microsoft.NETCore.App.Ref",
+        "microsoft.net.sdk.mono.emscripten.current" : "Microsoft.NET.Workload.Emscripten.Current"
+    }
+    rollback_dict: dict[str, str] = {}
+
+    # Load in the Version.Details.xml file
+    with open(os.path.join(get_repo_root_path(), "eng", "Version.Details.xml"), encoding="utf-8") as f:
+        version_details_xml = f.read()
+        root = ET.fromstring(version_details_xml)
+
+    # Get the General Band version from the Version.Details.xml file sdk version
+    general_version_obj = root.find(".//Dependency[@Name='VS.Tools.Net.Core.SDK.Resolver']")
+    if general_version_obj is not None:
+        full_band_version_holder = general_version_obj.get("Version", "ERROR: Failed to get version")
```

Comment:
I think it might be better to just do `get("Version")` then you can do `if full_band_version_holder is None`, since this string doesn't get rendered anywhere and it's just used for an equality check. Same for the other places that this is done.

Reply:
Good catch, didn't notice that default return value for get was None.